src/dbnode/storage/database.go

```diff
@@ -529,7 +529,16 @@ func (d *db) QueryIDs(
 		return index.QueryResults{}, err
 	}
 
-	return n.QueryIDs(ctx, query, opts)
+	var (
+		doneCh       = make(chan struct{})
```

Comment:
hm would using a waitGroup be a little cheaper here?

Reply:
oh yeah prolly, I'll fix